packages/server/graphql/private/types/SetIsFreeMeetingTemplateSuccess.ts

```diff
@@ -0,0 +1,19 @@
+import getRethink from '../../../database/rethinkDriver'
+import {SetIsFreeMeetingTemplateSuccessResolvers} from '../../private/resolverTypes'
+
+export type SetIsFreeMeetingTemplateSuccessSource = {
+  updatedTemplateIds: string[]
+}
+
+const SetIsFreeMeetingTemplateSuccess: SetIsFreeMeetingTemplateSuccessResolvers = {
+  updatedTemplates: async ({updatedTemplateIds}) => {
+    const r = await getRethink()
+    const updatedTemplates = await r
+      .table('MeetingTemplate')
+      .getAll(r.args(updatedTemplateIds))
+      .run()
+    return updatedTemplates
```

Comment:
+1 No need to change as this is not called often, but you could have just returned the `updatedTemplates` from `setIsFreeMeetingTemplate` directly instead of just the id and then reading it again here.

Reply:
I went with this approach as I was following a pattern that's used in some other mutations, e.g. [editTask](https://github.com/ParabolInc/parabol/blob/9177909d9b3bc9cf2ba5d88a32c0ce53b1da177f/packages/server/graphql/mutations/editTask.ts) has the `task` but returns the `taskId` so [EditTaskPayload](https://github.com/ParabolInc/parabol/blob/9177909d9b3bc9cf2ba5d88a32c0ce53b1da177f/packages/server/graphql/types/EditTaskPayload.ts) queries the task, or [flagConversionModel](https://github.com/ParabolInc/parabol/blob/fa767a6f719cec03a966d41cc45d1a077a1857fb/packages/server/graphql/private/mutations/flagConversionModal.ts) has the `organization` but returns the `orgId` so [FlagConversionModalPayload](https://github.com/ParabolInc/parabol/blob/ae11377bf1f529f954cbbe80064775b2e6182056/packages/server/graphql/private/types/FlagConversionModalPayload.ts) grabs the `organization`.  

I guess the difference is in `SetIsFreeMeetingTemplateSuccess`, we're querying directly to rethinkdb rather than using a dataLoader, so the other payloads will use the cache whereas this one won't.

Am I right in saying that the advantage of returning ids from the mutation rather than the object is because it adds an extra sanity check?

Reply:
Usually you should return ids to avoid overfetching, i.e. if the mutation did not fetch all the data that is needed for the success payload resolver, then it shouldn't fetch it just to return it. The mutation doesn't know[^1] what the client requested and should not fetch data which might not be needed.
`editTask` has a different use case with subscriptions, we don't want to serialize the data to send it over redis if it's not needed in the end. The return value could be different, but it's better to be the same for consistency, otherwise the success payload resolver would need to check for different inputs.
Why `flagConversionModal` does it this way, I don't know. I assume it is just to be aligned with the others.

[^1]: There is a 4th argument of type `GraphQLResolveInfo` to the resolver which contains this information, but it shouldn't really be used for this.

Reply:
Thanks! The overfetching and subscription use cases both make sense to me. 

For the [flagConversionModal](https://github.com/ParabolInc/parabol/blob/fa767a6f719cec03a966d41cc45d1a077a1857fb/packages/server/graphql/private/mutations/flagConversionModal.ts), @mattkrick it'd be great to know why you return the `id` rather than the `organization` object here?

Reply:
In a perfect world, we'd use dataloaders for reads & the writes wouldn't have to return anything (e.g. we wouldn't have to `RETURN *` in pg or `{returnChanges: true}` in rethinkdb.

in that scenario, you're still only issuing 1 write & 1 read, albeit the read is non-atomic & requires a separate round trip to the DB.

my reason for returning just an ID is to send as little data as possible through the data transform pipeline. we don't know which fields, if any, the user wants from the organization. so why send the whole org? it's not a huge issue either way, It's just simpler to reason about-- the SourceStream has the foregin keys, the ResopnseStream transforms that SourceStream into everything the client wants.